src/scenes/Partners/Detail/Tabs/Projects/PartnerDetailProjectsTable.tsx

```diff
@@ -1,93 +1,114 @@
 import { Box } from '@mui/material';
-import { DataGrid, DataGridProps, GridRowParams } from '@mui/x-data-grid';
+import {
+  DataGrid,
+  DataGridProps,
+  GridColDef,
+  GridRowParams,
+} from '@mui/x-data-grid';
 import { cmpBy, simpleSwitch } from '@seedcompany/common';
 import { useNavigate } from 'react-router-dom';
 import { ProjectStatusLabels } from '~/api/schema/enumLists';
 import { Sensitivity } from '~/api/schema/schema.graphql';
 import { labelFrom } from '~/common';
 import { SensitivityIcon } from '~/components/Sensitivity';
-import { PartnerDetailProjectsTableListItemFragment as Project } from './PartnerDetailProjectsTable.graphql';
+import {
+  PartnerDetailProjectsTableListItemFragment,
+  PartnerDetailProjectsTableListItemFragment as Project,
+} from './PartnerDetailProjectsTable.graphql';
 
 export interface PartnerDetailProjectsTableProps {
-  projects: readonly Project[];
-  GridProps?: Omit<DataGridProps<Project>, 'columns' | 'isCellEditable'>;
+  projects: readonly PartnerDetailProjectsTableListItemFragment[];
+  onPageChange?: (page: number) => void;
+  totalRows?: number;
+  GridProps?: Omit<
+    DataGridProps<Project>,
+    'columns' | 'rows' | 'isCellEditable'
+  >;
 }
 
 export const PartnerDetailProjectsTable = ({
   projects,
+  totalRows,
+  onPageChange,
 }: PartnerDetailProjectsTableProps) => {
   const navigate = useNavigate();
 
   const handleRowClick = (params: GridRowParams<Project>) => {
     navigate(`/projects/${params.row.id}`);
   };
 
+  const columns: Array<GridColDef<Project>> = [
+    {
+      headerName: 'Project Name',
+      field: 'name',
+      flex: 1,
+      valueGetter: ({ value }) => value.value,
+      renderCell: ({ value }) => (
+        <Box component="span" color="primary.main">
+          {value}
+        </Box>
+      ),
+    },
+    {
+      headerName: 'Status',
+      field: 'projectStatus',
+      flex: 1,
+      valueGetter: ({ value }) => labelFrom(ProjectStatusLabels)(value),
+    },
+    {
+      headerName: 'Engagements',
+      field: 'engagements',
+      flex: 1,
+      valueGetter: ({ value }) => value.total,
+    },
+    {
+      headerName: 'Sensitivity',
+      field: 'sensitivity',
+      flex: 1,
+      sortComparator: cmpBy<Sensitivity>((v) =>
+        simpleSwitch(v, { Low: 0, Medium: 1, High: 2 })
+      ),
+      renderCell: ({ value }) => (
+        <Box
+          display="flex"
+          alignItems="center"
+          gap={1}
+          textTransform="uppercase"
+        >
+          <SensitivityIcon value={value} disableTooltip />
+          {value}
+        </Box>
+      ),
+    },
+  ];
+
   return (
-    <DataGrid<Project>
+    <DataGrid
+      rows={projects}
+      columns={columns}
+      pageSize={25}
+      rowCount={totalRows ?? projects.length}
+      pagination
+      paginationMode="server"
```

Comment:
sorting also needs to be server side as well. Sorting a single page is kinda useless as it becomes a random subset of the unsorted entire list.

I don't believe the API supports sorting on `status` or _engagement total_ yet either. So that would need to be [added there](https://github.com/SeedCompany/cord-api-v3/blob/develop/src/components/project/project.repository.ts#L284-L289) as well.

Reply:
At the moment the internal team is not able to view all of the partner related projects because of the response limit. Should sorting be a separate ticket so that it does not block this fix?

Reply:
IMO no. If we are looking for a short term fix, I'd just set the `count: 100` and try to work around the pagination entirely.

Reply:
Alright. If someone from product mentions that this is emergent (they have not yet) we can just go that route as a temp band aid